recipes/glew/all/conanfile.py

```diff
@@ -68,18 +69,17 @@ def _configure_cmake(self):
         return cmake
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        files.apply_conandata_patches(self)
         cmake = self._configure_cmake()
         cmake.build()
 
     def package(self):
         self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.pdb")
+        files.rmdir(self, f"{self.package_folder}/lib/pkgconfig")
```

Comment:
did we change our preference for `os.path.join` or are the new functions smarter?

Reply:
Nope, keep os.path.join. Of course, "/" works too.

Reply:
my rationale is the following:

- It is safe to use a format string instead of os.path.join when all the arguments are known statically, and only the first one can contain an absolute path.
- `/` separator is properly interpreted on all platforms, so there is no bug
- using a format string is shorter to read/write than os.path.join
- it is clear to the reader that this is a path (call to `rmdir`, and `package_folder` variable)
- it allows to remove a python import

Reply:
I can't agree more.